gce/survival-training/README.md

```diff
@@ -0,0 +1,93 @@
+
+1. We could store the code in a [Cloud Source Repository](https://cloud.google.com/source-repositories/) and have the startup script clone it to the instance before doing anything else.
+
+In this guide, in the interests of flexibility, we will go with the Cloud Source Repository option. The reason this is attractive is that it lets you apply `git` semantics to your training jobs -- for example, you can [tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) different model architectures and specify them through the instance metadata. Or you could just use the trainer on some special branch of the repository which, again, would be specified at deployment time using instance metadata. This setup is much more powerful than the others.
+
+
+### Startup script
+
+GCE instances have an in-built mechanism that triggers a script on instance startup (for the appropriate definition of startup, but this need not concern us for now). This is the mechanism we will use to kick off a training job or to pick it up where it left off before a shutdown. We will make use of it through [startup scripts](https://cloud.google.com/compute/docs/startupscript#troubleshooting).
+
+It is alright if you are not familiar with these things. What is more important at the moment is to know that they exist and have a general sense for how they fit into the training process.
+
+
+### Instance metadata
+
+You may want to deploy different parametrizations of a training job to multiple GCE instances at different times. For this purpose, [we will make use of the metadata server available to every GCE instance](https://cloud.google.com/compute/docs/storing-retrieving-metadata#custom).
+
+Instance metadata is where we will store things like model hyperparameters and paths to training and evaluation data. We will store metadata on our instances before we start our training jobs, and our startup scripts will make use of this metadata on instance startup.
+
+If you are using the [example startup script provided in this guide](./gce/startup.sh), you will only have to make minimal changes to do so. The important thing again is to understand where instance metadata fits into our picture and that GCE guarantees that it will be available to our startup script when it runs.
+
+
+## Examples
+
+The following examples demonstrate the use of the framework provided here:
+
++ [TensorFlow estimator](./README-tf-estimator.md)
+
++ [sklearn model](./README-sklearn.md) - work in progress
+
+- - -
+
+##### Notes and references
+
++ The trainer CLI suggested here is modeled on the [the Cloud ML Engine trainer interface](https://cloud.google.com/ml-engine/docs/packaging-trainer). This has the sizable benefit that, should you ever require distribution of any TensorFlow models that you are training through this process on GCE, it should take very little work to move your training to ML Engine.
```

Comment:
> The trainer CLI suggested here is modeled on the [the Cloud ML Engine trainer interface](https://cloud.google.com/ml-engine/docs/packaging-trainer). This has the sizable benefit that, should you ever require distribution of any TensorFlow models that you are training through this process on GCE, it should take very little work to move your training to ML Engine.

The trainer command-line interface suggested here is modeled on the [the Cloud ML Engine trainer interface](https://cloud.google.com/ml-engine/docs/packaging-trainer). In case you want to perform distributed training with TensorFlow models, using this command-line interface for your trainer makes it easier to move your training from Compute Engine to Cloud ML Engine.